### PR TITLE
chore: [PL-43844]: implement no-external-snapshot custom eslint rule to prevent new external snapshot addition

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary
- We have pre-commit hook that prevents internal and new external snapshot addition. But. it can be bypassed by force-commit
- Implemented new eslint rule that does not allow new addition of external snapshots

**Note**: Adding new internal snapshot of more than 25 lines is already prevented by in-build eslint rule jest/no-large-snapshots

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Review by Eric761 — DISMISSED** [review]

**Review by shivanand-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

File: scripts/eslint-rules/no-external-snapshot.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+const { get } = require('lodash')
+
+const MAX_EXTERNAL_SNAPSHOT_COUNT = 3381 // These many times toMatchSnapshot is used as of Jan 3, 2024
```
Comment (abhinavrastogi-harness) [inline scripts/eslint-rules/no-external-snapshot.js:10]: hardcoding the max count will only for a week.. when someone removes one snapshot, this script will allow addition of one more snapshot again

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]
why do we need this change?
we can also just make `maxSize: 0`..that should prevent addition of snapshots directly.. and also prevent updates to existing snapshots (which will push for migration to assertions)

**Comment by chetanvekariya-harness** [comment]
> why do we need this change? we can also just make `maxSize: 0`..that should prevent addition of snapshots directly.. and also prevent updates to existing snapshots (which will push for migration to assertions)

That property (maxSize) does not work in the inbuilt rule. That is the reason we could not prevent external snapshots through lint rule earlier and had to write pre-commit hook instead.

File: scripts/eslint-rules/no-external-snapshot.js (hunk 1 above)
Comment (chetanvekariya-harness) [reply scripts/eslint-rules/no-external-snapshot.js:10]: Thanks for pointing this out. In that case, I should write something that checks quality like we discussed once. May be, no of snapshots / no of test cases or something similar.

**Comment by chetanvekariya-harness** [comment]
Closing this until I have proper solution.

**Pull request closed without merging**